lib/rubocop/cop/lint/suspicious_assignment.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for mistyped shorthand assignments.
+      #
+      # @example
+      #   # bad
+      #   x =- y
+      #   x =+ y
+      #   x =! y
+      #
+      #   # good
+      #   x -= y
+      #   x += y
+      #   x != y
+      #
+      class SuspiciousAssignment < Base
+        include RangeHelp
+        extend AutoCorrector
+
+        MSG = 'Suspicious assignment detected. Did you mean `%<op>s`?'
+
+        SIMPLE_ASSIGNMENT_TYPES = %i[lvasgn ivasgn cvasgn gvasgn casgn].freeze
+
+        MISTYPES = {
```

Comment:
I think something like `MISTYPED_ASSIGNMENTS" or "MISTAKES" would be a better name here.